### PR TITLE
Test for pure diffusion problem

### PR DESCRIPTION
**Pull request by sebglane**
After #124

### CODE REVIEW

**Comment by j507** [comment]
So, these are two tests problems in serial without using any of the classes/structs of our libraries? The goal is to validate adaptive time stepping? What is with the convergence rates? Some description for the PR would be nice.

File: tests/rotatingMHD/diffusion_fixed_step_size.output [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+cycle time_step cells dofs         L2               H1             Linfty      
+    0    0.5000 16384 16641 3.328e-02      - 2.958e-01      - 6.659e-02      - 
+    1    0.2500 16384 16641 1.861e-04 -14.96 1.654e-03 -14.97 3.723e-04 -14.97 
+    2    0.1250 16384 16641 5.654e-05  -3.44 5.027e-04  -3.44 1.131e-04  -3.44 
+    3    0.0625 16384 16641 1.176e-05  -4.53 1.068e-04  -4.47 2.356e-05  -4.53 
+    4    0.0312 16384 16641 2.690e-06  -4.26 3.325e-05  -3.37 5.430e-06  -4.23 
+
+cycle time_step cells dofs        L2              H1            Linfty      
+    0    0.5000 16384 16641 8.982e-03     - 7.982e-02     - 1.797e-02     - 
+    1    0.2500 16384 16641 9.461e-03  0.15 8.408e-02  0.15 1.893e-02  0.15
```
Comment (sebglane) [inline tests/rotatingMHD/diffusion_fixed_step_size.output:11]: Is something wrong here?

**Comment by sebglane** [comment]
> So, these are two tests problems in serial without using any of the classes/structs of our libraries? The goal is to validate adaptive time stepping? What is with the convergence rates? Some description for the PR would be nice.

Yes, the only class used is `RMHD::TimeDiscretization::VSIMEX`. These tests are the outcome of my debugging related to #118. The convergence rates for variable time stepping are not really meaningful, but the error decreases monotonically. To be honest, I don't know why the rate are equal -4 in case of fixed time stepping. I guess that this is a phenomenon related to the exponential decay of the solution. By implement some manufactured solution, we could double-check this. See also my comment to SBDF2 and fixed time stepping.

**Comment by j507** [comment]
> > So, these are two tests problems in serial without using any of the classes/structs of our libraries? The goal is to validate adaptive time stepping? What is with the convergence rates? Some description for the PR would be nice.
> 
> Yes, the only class used is `RMHD::TimeDiscretization::VSIMEX`. These tests are the outcome of my debugging related to #118. The convergence rates for variable time stepping are not really meaningful, but the error decreases monotonically. To be honest, I don't know why the rate are equal -4 in case of fixed time stepping. I guess that this is a phenomenon related to the exponential decay of the solution. By implement some manufactured solution, we could double-check this. See also my comment to SBDF2 and fixed time stepping.

The rate of convergence should be -2. I do not see how this would be different in this case. Neither on `TGV.cc` nor `ThermalTGC.cc` we got such results. 
I would prefer to avoid convergence tests mirroring the Taylor-Green vortex "closed" solution and instead resort to manufactured solutions with a right-hand side.

File: tests/rotatingMHD/diffusion_fixed_step_size.output (hunk 1 above)
Comment (j507) [reply tests/rotatingMHD/diffusion_fixed_step_size.output:11]: Not only that but the magnitude of the convergence rates are off. A second order scheme is used, so second order or lower would be expected.

**Comment by sebglane** [comment]
> > > So, these are two tests problems in serial without using any of the classes/structs of our libraries? The goal is to validate adaptive time stepping? What is with the convergence rates? Some description for the PR would be nice.
> > 
> > 
> > Yes, the only class used is `RMHD::TimeDiscretization::VSIMEX`. These tests are the outcome of my debugging related to #118. The convergence rates for variable time stepping are not really meaningful, but the error decreases monotonically. To be honest, I don't know why the rate are equal -4 in case of fixed time stepping. I guess that this is a phenomenon related to the exponential decay of the solution. By implement some manufactured solution, we could double-check this. See also my comment to SBDF2 and fixed time stepping.
> 
> The rate of convergence should be -2. I do not see how this would be different in this case. Neither on `TGV.cc` nor `ThermalTGC.cc` we got such results.
> I would prefer to avoid convergence tests mirroring the Taylor-Green vortex "closed" solution and instead resort to manufactured solutions with a right-hand side.



> > > So, these are two tests problems in serial without using any of the classes/structs of our libraries? The goal is to validate adaptive time stepping? What is with the convergence rates? Some description for the PR would be nice.
> > 
> > 
> > Yes, the only class used is `RMHD::TimeDiscretization::VSIMEX`. These tests are the outcome of my debugging related to #118. The convergence rates for variable time stepping are not really meaningful, but the error decreases monotonically. To be honest, I don't know why the rate are equal -4 in case of fixed time stepping. I guess that this is a phenomenon related to the exponential decay of the solution. By implement some manufactured solution, we could double-check this. See also my comment to SBDF2 and fixed time stepping.
> 
> The rate of convergence should be -2. I do not see how this would be different in this case. Neither on `TGV.cc` nor `ThermalTGC.cc` we got such results.
> I would prefer to avoid convergence tests mirroring the Taylor-Green vortex "closed" solution and instead resort to manufactured solutions with a right-hand side.

Could you implement such a test? I did the ones above in order to verify whether we have convergence at... Therefore, I was kind of happy with the -4.

**Comment by sebglane** [comment]
@j507 Could you let me know whether you will implement a manufactured solution in these tests?

**Comment by j507** [comment]
> @j507 Could you let me know whether you will implement a manufactured solution in these tests?

Yes, I will redo this tests with a manufactured solution.

**Review by j507 — APPROVED** [review]
I will change the tests in a different PR

**Pull request merged**